src/components/RandomPicker/manifest.json

```diff
@@ -9,98 +9,85 @@
   "props": [
     {
       "name": "listOfDataSource",
-      "displayName": "Data Source",
+      "displayName": "Collection",
       "helpText": "One of the items will be picked up randomly.",
       "type": "list"
     },
     {
       "name": "itemOfDataSourceList",
-      "displayName": "Item of Data Source",
-      "helpText": "You can concatenate items as you like.",
+      "displayName": "Display Text",
       "type": "text",
       "role": "listItem",
       "reference": "listOfDataSource"
     },
     {
-      "name": "enableNotFoundText",
-      "displayName": "Set a default Not Found text?",
-      "type": "boolean",
-      "default": false
+      "name": "actionType",
+      "displayName": "What happens when the user clicks on the component?",
+      "type": "number",
+      "default": 10,
+      "control": {
+        "type": "menu",
+        "options": [
+          { "label": "Trigger Click Actions", "value": 10 },
+          { "label": "Refresh the component", "value": 20 }
+        ]
+      }
     },
     {
-      "name": "notFoundText",
-      "displayName": "Not Found Text",
-      "helpText": "Set a default 'Not Found' text when there is no available item in your list. (Default: 'Not Found')",
-      "type": "text",
+      "name": "clickActions",
+      "displayName": "Click Actions",
+      "helpText": "The actions will be triggered when the component is clicked.",
+      "type": "action",
+      "role": "listItem",
+      "reference": "listOfDataSource",
+      "arguments": [
+        {
+          "type": "text",
+          "displayName": "Random Item on Click"
+        }
+      ],
       "enabled": {
-        "enableNotFoundText": true
+        "actionType": 10
       }
     },
     {
-      "name": "enablePreviewDemoText",
-      "displayName": "Set a default Preview Demo text?",
+      "name": "enableRemoveDuplicatesOnRefresh",
+      "displayName": "Remove duplicates?",
+      "helpText": "Remove the item already shown from the list. When you refresh or move to a screen, it will reset all values.",
       "type": "boolean",
-      "default": false
+      "default": false,
+      "enabled": {
+        "actionType": 20
+      }
     },
     {
-      "name": "previewDemoText",
-      "displayName": "Preview Demo Text",
-      "helpText": "Set a Preview Demo text. This is only used in editor. (Default: 'Preview demo')",
+      "name": "introText",
+      "displayName": "Intro Text",
+      "helpText": "Set an intro message on the component before clicking. You can skip this message if you leave it as empty.",
       "type": "text",
+      "default": "Click to Start!",
       "enabled": {
-        "enablePreviewDemoText": true
+        "actionType": 20
+      }
+    },
+    {
+      "name": "refreshActions",
+      "displayName": "Actions after refresh",
+      "helpText": "The actions will be triggered after the component is refreshed.",
+      "type": "action",
+      "reference": "listOfDataSource",
```

Comment:
Since you've added reference here, you may need to add role as well!

Reply:
I don't think it's a good idea since it's very flaky when I use reference & role here due to a timing issue. So I've deleted the reference to avoid unexpected behavior.